manifests/snmp.pp

```diff
@@ -10,7 +10,7 @@
   String $snmp         = 'public',
   Integer $lan_channel = 1,
 ) {
-  require ipmi
+  require ipmi::config
```

Comment:
`ipmi::config` needs the base class to be evaluated to provide `$ipmi::watchdog`. What are you trying to accomplish by not requiring the base class?